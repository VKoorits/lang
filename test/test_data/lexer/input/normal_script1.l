nums = [ 1->60 ] # nums = [1,2,3,...,59,60]
for i in nums:
	for j in nums:
		if random.float() > 0.5 :
			swap i, j
/*
	bubble sort, 
	but in random order
*/
print nums
_var1 = "say \"cheese'"
_var1 = 'say "cheese\''
_var1 = 'say \'cheese"'
_var1 = "say 'cheese\""

_var2 = 159_267_843.000_000_1
