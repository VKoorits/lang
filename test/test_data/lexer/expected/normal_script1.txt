IDENT_TOKEN: nums, info => 0
OPERATION_TOKEN: =, info => 1
OPERATION_TOKEN: [, info => -1
INT_NUM_TOKEN: 1, info => 0
OPERATION_TOKEN: ->, info => -2
INT_NUM_TOKEN: 60, info => 0
OPERATION_TOKEN: ], info => -1
#
IDENT_TOKEN: for, info => 0
IDENT_TOKEN: i, info => 0
OPERATION_TOKEN: in, info => 13
IDENT_TOKEN: nums, info => 0
OPERATION_TOKEN: :, info => -2
#
IDENT_TOKEN: for, info => 0
IDENT_TOKEN: j, info => 0
OPERATION_TOKEN: in, info => 13
IDENT_TOKEN: nums, info => 0
OPERATION_TOKEN: :, info => -2
#
IDENT_TOKEN: if, info => 1
IDENT_TOKEN: random, info => 0
OPERATION_TOKEN: ., info => 14
IDENT_TOKEN: float, info => 0
OPERATION_TOKEN: (, info => -1
OPERATION_TOKEN: ), info => -1
OPERATION_TOKEN: >, info => 7
FLOAT_NUM_TOKEN: 0.5, info => 0
OPERATION_TOKEN: :, info => -2
#
IDENT_TOKEN: swap, info => 0
IDENT_TOKEN: i, info => 0
OPERATION_TOKEN: ,, info => -2
IDENT_TOKEN: j, info => 0
#
IDENT_TOKEN: print, info => 0
IDENT_TOKEN: nums, info => 0
#
IDENT_TOKEN: _var1, info => 0
OPERATION_TOKEN: =, info => 1
STRING_TOKEN: say "cheese', info => 0
#
IDENT_TOKEN: _var1, info => 0
OPERATION_TOKEN: =, info => 1
STRING_TOKEN: say "cheese', info => 0
#
IDENT_TOKEN: _var1, info => 0
OPERATION_TOKEN: =, info => 1
STRING_TOKEN: say 'cheese", info => 0
#
IDENT_TOKEN: _var1, info => 0
OPERATION_TOKEN: =, info => 1
STRING_TOKEN: say 'cheese", info => 0
#
IDENT_TOKEN: _var2, info => 0
OPERATION_TOKEN: =, info => 1
FLOAT_NUM_TOKEN: 159267843.0000001, info => 0
#
47
